src/nspawn/nspawn.c

```diff
@@ -1537,6 +1541,16 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_settings_mask |= SETTING_BIND_USER;
                         break;
 
+                case ARG_BIND_USER_SHELL:
+                        r = parse_bind_user_shell(optarg, &arg_bind_user_shell);
```

Comment:
If the command line option is specified twice, then the previous assignment is leaked.
Please introduce a temporal variable, then use `free_and_replace()` on success.